src/main/java/com/minecolonies/coremod/util/ItemStackUtils.java

```diff
@@ -0,0 +1,391 @@
+        {
+            return -1;
+        }
+        //todo: use 'better' version of this thing
+        if (ToolType.HOE.equals(toolType))
+        {
+             if (stack.getItem() instanceof ItemHoe)
+             {
+                 final ItemHoe itemHoe = (ItemHoe)stack.getItem();
+                 return getToolLevel(itemHoe.getMaterialName());
+             }
+        }
+        else if (ToolType.SWORD.equals(toolType))
+        {
+            if (stack.getItem() instanceof ItemSword)
+            {
+                final ItemSword itemSword = (ItemSword)stack.getItem();
+                return getToolLevel(itemSword.getToolMaterialName());
+            }
+        }
+        else if (!toolType.hasVariableMaterials())
+        {
+            //We need a hut level 1 minimum
+            return 1;
+        }
+        else
+        {
+            return stack.getItem().getHarvestLevel(stack, toolType.getName(), null, null);
+        }
+        return -1;
+    }
+
+
+    /**
+     * Verifies if an item has an appropriated grade.
+     *
+     * @param itemStack    the type of tool needed
+     * @param toolLevel    the tool level
+     * @param minimalLevel the minimum level needed
+     * @param maximumLevel the maximum level needed (usually the worker's hut level)
+     * @return true if tool is acceptable
+     */
+    public static boolean verifyToolLevel(@NotNull final ItemStack itemStack, final int toolLevel, final int minimalLevel, final int maximumLevel)
+    {
+        if (toolLevel < minimalLevel)
+        {
+            return false;
+        }
+        return (toolLevel + getMaxEnchantmentLevel(itemStack) <= maximumLevel);
+    }
+
+    /**
+     * Calculates the fortune level this tool has.
+     *
+     * @param tool the tool to check.
+     * @return fortune level.
+     */
+    public static int getFortuneOf(@Nullable final ItemStack tool)
+    {
+        if (tool == null)
+        {
+            return 0;
+        }
+        //calculate fortune enchantment
+        int fortune = 0;
+        if (tool.isItemEnchanted())
+        {
+            final NBTTagList t = tool.getEnchantmentTagList();
+
+            for (int i = 0; i < t.tagCount(); i++)
+            {
+                final int id = t.getCompoundTagAt(i).getShort(NBT_TAG_ENCHANT_ID);
+                if (id == FORTUNE_ENCHANT_ID)
+                {
+                    fortune = t.getCompoundTagAt(i).getShort(NBT_TAG_ENCHANT_LEVEL);
+                }
+            }
+        }
+        return fortune;
+    }
+
+    /**
+     * Calculates the max level enchantment this tool has.
+     *
+     * @param itemStack the tool to check.
+     * @return max enchantment level.
+     */
+    public static int getMaxEnchantmentLevel(final ItemStack itemStack)
+    {
+        if (itemStack == null)
+        {
+            return 0;
+        }
+        int maxLevel = 0;
+        if (itemStack != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)